app/actions/BarsLockoutActionFilter.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package actions
+
+import actionsmodel.AuthenticatedJourneyRequest
+import controllers.{JourneyIncorrectStateRouter, routes}
+import essttp.bars.BarsVerifyStatusConnector
+import essttp.journey.model.Journey
+import play.api.Logging
+import play.api.mvc.{ActionFilter, Request, Result, Results}
+import util.QueryParameterUtils._
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class BarsLockoutActionFilter @Inject() (barsVerifyStatusConnector: BarsVerifyStatusConnector)(
+    implicit
+    ec: ExecutionContext
+) extends ActionFilter[AuthenticatedJourneyRequest] with Logging with Results {
+
+  override protected def filter[A](journeyRequest: AuthenticatedJourneyRequest[A]): Future[Option[Result]] = {
+    implicit val rh: Request[A] = journeyRequest.request
+
+    journeyRequest.journey match {
+      case j: Journey.BeforeComputedTaxId =>
+        JourneyIncorrectStateRouter.logErrorAndRouteToDefaultPageF(j).map(Some(_))
+      case j: Journey.AfterComputedTaxId =>
+        barsVerifyStatusConnector.status(j.taxId).map { status =>
+          status.lockoutExpiryDateTime.map { expiresAt =>
+            logger.info(s"BARs lockout for taxId: ${j.taxId}")
```

Comment:
We may want to change the log level, will this log tax ids in prod?

Reply:
Yep. @AndyHWChung Not sure if this is an issue though. Happy to take out. We may be expected to audit this soon in any case.

Reply:
I don't think we're allowed to put any taxids into kibana

Reply:
tax id's themselves are tolerated in kibana (see [here](https://confluence.tools.tax.service.gov.uk/x/D4RxHQ)) so I think this would be safe if we really wanted it